pkgs/misc/emulators/citra/base.nix

```
@@ -0,0 +1,195 @@
+, sndioSupport     ? true,                                sndio
+, udevSupport      ? stdenv.isLinux,                      udev
+}:
+
+with lib;
+let
+  cpp-jwt-src = fetchFromGitHub {
+    owner = "arun11299";
+    repo = "cpp-jwt";
+    rev = "6e27aa4c8671e183f11e327a2e1f556c64fdc4a9";
+    sha256 = "1mqcgyp9sm846n8za85i18hgw2x0z8f0npdgfpd9gykzwf72mqmv";
+  };
+  cryptopp-src = fetchFromGitHub {
+    owner = "weidai11";
+    repo = "cryptopp";
+    rev = "f320e7d92a33ee80ae42deef79da78cfc30868af";
+    sha256 = "19sxiwzbr82zhjm36qd2p1zx710ilzgvj9945yd5mz7bvzpbpw5q";
+  };
+  cubeb-src = fetchFromGitHub {
+    owner = "mozilla";
+    repo = "cubeb";
+    rev = "1d66483ad2b93f0e00e175f9480c771af90003a7";
+    sha256 = "06rlmdarrk9hwm8svmh2nl2cd4ybmzm2jhp7ih6xzzrba31hg7n3";
+    fetchSubmodules = true;
+  };
+  dynarmic-src = fetchFromGitHub {
+    owner = "citra-emu";
+    repo = "dynarmic";
+    rev = "af0d4a7c18ee90d544866a8cf24e6a0d48d3edc4";
+    sha256 = "1hsl8vdd2y2h2rmhx3bvh4a8l3m33xnsnk5v64jkwdfcl15iyh2a";
+  };
+  lodepng-src = fetchFromGitHub {
+    owner = "lvandeve";
+    repo = "lodepng";
+    rev = "31d9704fdcca0b68fb9656d4764fa0fb60e460c2";
+    sha256 = "1il97cq1rms3078zsyq00acgb75kva1smcrkb8r450p6c1rmwsii";
+  };
+  nihstro-src = fetchFromGitHub {
+    owner = "neobrain";
+    repo = "nihstro";
+    rev = "fd69de1a1b960ec296cc67d32257b0f9e2d89ac6";
+    sha256 = "1lfx92lpszmz1xsvb80a349r3i7g5msxixkb29v1qvi4a8v3ak58";
+  };
+  soundtouch-src = fetchFromGitHub {
+    owner = "citra-emu";
+    repo = "ext-soundtouch";
+    rev = "060181eaf273180d3a7e87349895bd0cb6ccbf4a";
+    sha256 = "0x92a5wb2hcaqzckpdqsggbf863a9kicpa4xz4g4gdyid19n3ci9";
+  };
+  teakra-src = fetchFromGitHub {
+    owner = "wwylele";
+    repo = "teakra";
+    rev = "3e032a73d7e97eb434a053391d95029eebd7e189";
+    sha256 = "1cpxj8jvkj52f3jpy7rsgzs09h8cd3jmx6f2282dh9w2daaga4w8";
+  };
+  xbyak-src = fetchFromGitHub {
+    owner = "herumi";
+    repo = "xbyak";
+    rev = "c306b8e5786eeeb87b8925a8af5c3bf057ff5a90";
+    sha256 = "13l5yyaf67h6s0zlzksah4rybv4xsl1a7939frvwahxch88wfslr";
+  };
+  # Last updated Thu, 28 Oct 2021 12:43:04 GMT from https://api.citra-emu.org/gamedb/
+  compat-src = ./compatibility_list.json;
+in stdenv.mkDerivation {
+  inherit pname version src;
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=OFF"
+    "-DCITRA_USE_BUNDLED_SDL2=OFF"
+    "-DCITRA_USE_BUNDLED_QT=OFF"
+    "-DCITRA_USE_BUNDLED_FFMPEG=OFF"
+    "-DDYNARMIC_ENABLE_CPU_FEATURE_DETECTION=OFF"
+    "-DENABLE_COMPATIBILITY_LIST_DOWNLOAD=OFF" # Sacrifices reproducibility
+    "-DCITRA_ENABLE_COMPATIBILITY_REPORTING=ON"
+    "-DENABLE_FFMPEG_VIDEO_DUMPER=ON"
+    "-DENABLE_QT_TRANSLATION=ON"
+    "-DUSE_SYSTEM_BOOST=ON"
+    "-GNinja"
+    "-Wno-dev"
+    "-DUSE_DISCORD_PRESENCE=${if discordSupport then "ON" else "OFF"}"
+    "-DENABLE_WEB_SERVICE=${if onlineSupport then "ON" else "OFF"}"
+    (if fdkSupport     then "-DENABLE_FDK=ON"           else "-DENABLE_FFMPEG_AUDIO_DECODER=ON")
```

For consistency with the comments above
```suggestion
    (if fdkSupport then "-DENABLE_FDK=ON" else "-DENABLE_FFMPEG_AUDIO_DECODER=ON")
```

---

Thanks!